src/types.rs

```diff
@@ -1469,43 +1612,103 @@ impl Rule {
             false
         }
     }
-    /// Construct a rewrite rule from a left-hand-side (LHS) [`Term`] with one
-    /// or more right-hand-side (RHS) [`Term`]s. Returns `None` if the rule is
+    /// Construct a rewrite `Rule` from a left-hand-side (LHS) [`Term`] with one
+    /// or more right-hand-side (RHS) [`Term`]s. Returns `None` if the `Rule` is
     /// not valid.
     ///
     /// Valid rules meet two conditions:
     ///
     /// 1. `lhs` is an [`Application`]. This prevents a single rule from
-    ///    matching all possible terms
+    ///    matching all possible [`Term`]s
     /// 2. A [`Term`] in `rhs` can only use a [`Variable`] if it appears in
     ///    `lhs`. This prevents rewrites from inventing arbitrary terms.
     ///
     /// [`Term`]: enum.Term.html
     /// [`Application`]: enum.Term.html#variant.Application
     /// [`Variable`]: struct.Variable.html
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// # use term_rewriting::{Signature, Term, Rule, parse_term, parse_rule};
+    /// let mut sig = Signature::default();
+    /// 
+    /// let r = parse_rule(&mut sig, "A = B").expect("parse of A = B");
+    ///
+    /// let lhs = parse_term(&mut sig, "A").expect("parse of A");
+    /// let rhs = vec![parse_term(&mut sig, "B").expect("parse of B")];
+    /// let r2 = Rule::new(lhs, rhs).unwrap();
+    ///
+    /// assert_eq!(r, r2);
+    /// ```
```

Comment:
It looks like you're using `Rule::new` throughout, which is great! Because `parse_rule` could be doing just about anything behind the scenes, this might be a good place to also use the `Rule { lhs, rhs }` syntax. Seeing that the unwrapped `r2` is equal not only to a parsed rule but also to `Rule { lhs, rhs }` might help people ground out what `Rule::new` is doing.